lib/resty/healthcheck.lua

```diff
@@ -922,12 +945,16 @@ end
 -- Event handler callback
 function checker:event_handler(event_name, ip, port, hostname)
 
-  local target_found = (self.targets[ip] or EMPTY)[port]
+  local target_found = get_target(self, ip, port, hostname)
 
   if event_name == self.events.remove then
     if target_found then
       -- remove hash part
-      self.targets[target_found.ip][target_found.port] = nil
+      self.targets[target_found.ip][target_found.port][target_found.hostname] = nil
+      if not next(self.targets[target_found.ip][target_found.port]) then
+        -- no more hostnames on this port, so delete it
+        self.targets[target_found.ip][target_found.port] = nil
+      end
```

Comment:
slightly below in the
```lua
-- remove from list part
```
we do not include the check on hostname. Since that is a data-integrity thing, that should definitely have a test

Reply:
alternatively add the validation in the code. Add something like:
```lua
<loop start>
   if xyz then
     found = true
     break
  end
end
assert(found, "entry not found in numerical part")
```

Does that make sense?

Reply:
Good catch, thanks! I added the test for the hostname.